### PR TITLE
Tests for user settings.

### PR DESCRIPTION
**Pull request by besentv**
Signed-off-by: Bernhard Kölbl <besentv@gmail.com>

### CODE REVIEW

**Comment by besentv** [comment]
I'll merge the tests into the mono test code when I'm back from vaction.

**Comment by besentv** [comment]
Reopening because integrating the test into the mono test suite is pain

File: tools/tests/configuration-with-defaults.cs [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+		public static int Main(string[] args)
+		{
+			int i = 0;
+			string[] varNames = new string[] { "SettingBool2",  "SettingInt1", "SettingInt2", "SettingCustomClass1" };
+			string[] varValues = new string[] { "False", "4001", "4002", "abcdefghijklmno" };
+
+			Console.WriteLine($"SettingBool1: {Settings.Default.SettingBool1}, " +
+							  $"SettingBool2: {Settings.Default.SettingBool2}, " +
+							  $"SettingInt1: {Settings.Default.SettingInt1}, " +
+							  $"SettingBool2: {Settings.Default.SettingBool2}, " +
```
Comment (madewokherd) [inline tools/tests/configuration-with-defaults.cs:47]: Seems like this is printing the same value twice?

**Comment by besentv** [comment]
I'll do another change

**Comment by madewokherd** [comment]
Merged manually so I could also update the expected results.

**Pull request closed without merging**